File: tests/grader_language_tools_test_problems/hworld/solutions/john-js/hworld.js
"use strict";

var fs = require("fs");

var readInput = function(fileName) {
    var fileLines = fs.readFileSync(fileName).toString().split("\n");
    var count = parseInt(fileLines[0], 10);
    return count;
};

var solve = function(count) {
    return Array(count).fill("Hello world!");
};

var writeOutput = function(fileName, lines) {
    fs.writeFileSync(fileName, lines.join("\n"));
};

var inputFileName = process.argv[2] || "hworld.in";
var outputFileName = process.argv[3] || "hworld.out";

var count = readInput(inputFileName);
var lines = solve(count);
writeOutput(outputFileName, lines);

File: tests/grader_verification_test_problems/hworld/solutions/charlie-cpp-w/hworld.cpp
#include <iostream>
#include <fstream>
#include <vector>

int readInput(std::string fileName) {
    std::ifstream inputFile(fileName, std::ifstream::in);
    
    int count;
    inputFile >> count;
    inputFile.close();

    return count;
}

std::vector<std::string> solve(int count) {
    std::vector<std::string> lines(count, "Hello World");
    return lines;
}

void writeOutput(std::string fileName, std::vector<std::string> lines) {
    std::ofstream outputFile(fileName, std::ifstream::out);
    for (std::string line: lines) {
        outputFile << line << std::endl;
    }
    outputFile.close();
}

int main(int argc, const char * argv[]) {
    auto count = readInput("hworld.in");
    auto lines = solve(count);
    writeOutput("hworld.out", lines);
    return 0;
}
